e2e/app/page/admin-user-profile-info.ts

```diff
@@ -0,0 +1,172 @@
+      name: LabelAlias.VerifiedInstitution,
+      ancestorLevel: 2
+    }
+  },
+  InstitutionalRoleSelect: {
+    textOption: {
+      type: ElementType.Dropdown,
+      name: LabelAlias.InstitutionalRole,
+      ancestorLevel: 2
+    }
+  }
+};
+
+export default class UserProfileInfo extends AuthenticatedPage {
+  constructor(page: Page) {
+    super(page);
+  }
+
+  async isLoaded(): Promise<boolean> {
+    await Promise.all([waitForDocumentTitle(this.page, PageTitle)]);
+    return true;
+  }
+
+  getSaveButton(): Button {
+    return Button.findByName(this.page, { name: LinkText.Save });
+  }
+
+  async waitForSaveButton(isActive: boolean): Promise<Button> {
+    const button = this.getSaveButton();
+    const isCursorEnabled = !(await button.isCursorNotAllowed());
+    expect(isCursorEnabled).toBe<boolean>(isActive);
+    return button;
+  }
+
+  getAccountAccessToggle(): Checkbox {
+    const xpath = "//label[text()='Account access']/following-sibling::label/div/input[@type='checkbox']";
+    return new Checkbox(this.page, xpath);
+  }
+
+  // extract only the Account access text to verify the user is enabled or disabled
+  async getAccountAccessText(): Promise<string> {
+    const xpath = "//label[text()='Account access']/following-sibling::label/span";
+    const element = BaseElement.asBaseElement(this.page, await this.page.waitForXPath(xpath, { visible: true }));
+    const textContent = element.getTextContent();
+    return textContent;
+  }
+
+  getNameInput(): Textbox {
+    return Textbox.findByName(this.page, { dataTestId: DataTestIdAlias.UserFullName });
+  }
+
+  getUsernameInput(): Textbox {
+    return Textbox.findByName(this.page, { dataTestId: DataTestIdAlias.Username });
+  }
+
+  // get the placeholder in username field
+  getUserNamePlaceHolder(): Promise<string> {
+    const userName = this.getUsernameInput();
+    const userProfileEmail = userName.getProperty<string>('placeholder');
+    return userProfileEmail;
+  }
+
+  getContactEmailInput(): Textbox {
+    return Textbox.findByName(this.page, { dataTestId: DataTestIdAlias.ContactEmail });
+  }
+
+  getFreeCreditsUsedInput(): Textbox {
+    return Textbox.findByName(this.page, { dataTestId: DataTestIdAlias.FreeCreditsUsed });
+  }
+
+  // get the free credits used value
+  async getFreeCreditMaxValue(): Promise<string> {
+    const freeCreditsMax = this.getFreeCreditsUsedInput();
+    const freeCreditsMaxValue = await freeCreditsMax.getProperty<string>('value');
+    const regex = new RegExp(/(?<=of ).*(?= limit)/);
+    return regex.exec(freeCreditsMaxValue)[0];
+  }
+
+  // get the free credits limit value
+  async getFreeCreditsLimitValue(): Promise<string> {
+    const freeCreditsDropdown = SelectMenu.findByName(this.page, { dataTestId: DataTestIdAlias.FreeCreditLimit });
+    return await freeCreditsDropdown.getSelectedValue();
+  }
+
+  // get each modules bypass access status
+  async getEachBypassStatus(toggleAccessText: string): Promise<boolean> {
```

Comment:
Hi Joel & Alex, this is the generic function that will help get the module-toggle status individually.

Reply:
👍🏻